### PR TITLE
Fix/Refactor pipelined stateless restart

### PR DESCRIPTION
**Pull request by koolzz**
Signed-off-by: Nick Yurchenko <koolzz@fb.com>

## Summary
 - Cleanup the functionality
 - Enable inout stateless mode

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by uri200 — APPROVED** [review]

**Review by themarwhal — APPROVED** [review]

File: lte/gateway/python/magma/pipelined/app/inout.py [hunk 1]
```diff
@@ -155,51 +154,34 @@ def _get_config(self, config_dict):
...
-        ret[self._midle_tbl_num] = remaining_flows
-        if msgs:
-            chan = self._msg_hub.send(msgs, datapath)
-            self._wait_for_responses(chan, len(msgs))
-
-        return ret
+
+        if self.config.setup_type == 'XWF':
+            self.delete_all_flows(datapath)
+            self._install_default_flows(datapath)
```
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/app/inout.py:160]: XWF flows are not handled by restart controller? can you mark it as todo.

File: lte/gateway/python/magma/pipelined/tests/snapshots/test_inout.InOutTestXWF.testFlowSnapshotMatch.snapshot [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ cookie=0x0, table=ingress(main_table), n_packets=0, n_bytes=0, priority=10,in_port=LOCAL actions=set_field:0x10->reg1,resubmit(,middle(main_table)),set_field:0->reg0,set_field:0->reg3
+ cookie=0x0, table=ingress(main_table), n_packets=0, n_bytes=0, priority=0 actions=set_field:0x1->reg1,resubmit(,middle(main_table)),set_field:0->reg0,set_field:0->reg3
+ cookie=0x0, table=middle(main_table), n_packets=0, n_bytes=0, priority=15,reg6=0x1 actions=resubmit(,egress(main_table)),set_field:0->reg0,set_field:0->reg3
+ cookie=0x0, table=middle(main_table), n_packets=0, n_bytes=0, priority=10 actions=resubmit(,egress(main_table)),set_field:0->reg0,set_field:0->reg3
+ cookie=0x0, table=egress(main_table), n_packets=0, n_bytes=0, priority=0,reg1=0x10 actions=output:32768
+ cookie=0x0, table=egress(main_table), n_packets=0, n_bytes=0, priority=0,reg1=0x1 actions=LOCAL
```
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/tests/snapshots/test_inout.InOutTestXWF.testFlowSnapshotMatch.snapshot:6]: when sending packet to LOCAL, dont we need to set dst mac address? ref: _get_default_egress_flow_msgs()

File: lte/gateway/python/magma/pipelined/app/inout.py (hunk 1 above)
Comment (koolzz) [reply lte/gateway/python/magma/pipelined/app/inout.py:160]: yeah XWF doesn't run sessiond and we don't currently support stateless without it. I'll add a TODO

File: lte/gateway/python/magma/pipelined/tests/snapshots/test_inout.InOutTestXWF.testFlowSnapshotMatch.snapshot (hunk 2 above)
Comment (koolzz) [reply lte/gateway/python/magma/pipelined/tests/snapshots/test_inout.InOutTestXWF.testFlowSnapshotMatch.snapshot:6]: doesn't seem to be happening in other snapshots as well so I'm assuming its expected behavior

**Review by pshelar — APPROVED** [review]

**Pull request merged**